src/core/js/views/buttonsView.js

```diff
@@ -91,12 +95,14 @@ define(function() {
                 }
 
             } else {
-                // Backwords compatibility with v1.x
-                var ariaLabel = this.model.get('_buttons')["_" + changedAttribute].ariaLabel;
-                var buttonText = this.model.get('_buttons')["_" + changedAttribute].buttonText;
 
-                switch (changedAttribute) {
-                    case "showCorrectAnswer": case "hideCorrectAnswer":
+                // Backwords compatibility with v1.x               
+                var ariaLabel = this.model.get('_buttons')["_" + buttonState.asString].ariaLabel;
```

Comment:
`buttonState.asString` won't suffice – needs underscores removing and converting to camel case.